Mathlib/Probability/Integration.lean

```diff
@@ -127,6 +127,27 @@ theorem lintegral_mul_eq_lintegral_mul_lintegral_of_indepFun'' (h_meas_f : AEMea
     ∫⁻ ω, f ω * g ω ∂μ = (∫⁻ ω, f ω ∂μ) * ∫⁻ ω, g ω ∂μ :=
   lintegral_mul_eq_lintegral_mul_lintegral_of_indepFun' h_meas_f h_meas_g h_indep_fun
 
+theorem lintegral_fin_prod_eq_lintegral_fin_prod_lintegral_of_indepFun {ι : Type*} [DecidableEq ι]
+    [IsProbabilityMeasure μ] (s : Finset ι) (X : ι → Ω → ENNReal)
+    (hX : iIndepFun (fun _ ↦ ENNReal.measurableSpace) X μ)
+    (x_mea : ∀ i, Measurable (X i)) :
+    ∫⁻ (ω : Ω), ∏ i ∈ s, (X i ω) ∂μ = ∏ i ∈ s, ∫⁻ (ω : Ω), X i ω ∂μ := by
+  induction s using Finset.induction
+  case empty => simp only [Finset.prod_empty, lintegral_const, measure_univ, mul_one]
+  case insert _ _ j s hj v =>
+    calc  ∫⁻ (ω : Ω), ∏ i ∈ insert j s, X i ω ∂μ
+      _ = ∫⁻ (ω : Ω), (∏ i ∈ insert j s, X i) ω ∂μ := by simp
+      _ =  ∫⁻ (ω : Ω), (X j * ∏ i ∈ s, X i) ω ∂μ :=
+        lintegral_congr fun ω ↦ congrFun (Finset.prod_insert hj) ω
+      _ = (∫⁻ (ω : Ω), X j ω ∂μ) * ∫⁻ (ω : Ω), (∏ i ∈ s, X i) ω ∂μ := by
+        apply lintegral_mul_eq_lintegral_mul_lintegral_of_indepFun'
+        · exact Measurable.aemeasurable (x_mea j)
+        · apply s.aemeasurable_prod' (fun i _ ↦ by exact Measurable.aemeasurable (x_mea i))
```

Comment:
In the vast majority of cases `by exact` does nothing (`by` enters tactic mode and `exact` switches back to term mode).
```suggestion
        · exact s.aemeasurable_prod' (fun i _ ↦ (x_mea i).aemeasurable)
```